packages/react/src/state.tsx

```diff
@@ -0,0 +1,98 @@
+import { createContext, useReducer } from 'react'
+
+import { action } from './events'
+import { ActionDataState, JSONValue, StateDataState } from './template'
+
+export type LocalState = Record<string, JSONValue>
+
+type StateUpdate<T> = T | StateModifier
+type UpdateStateAction<T> = ActionDataState<['state-update', StateDataState<T>, StateUpdate<T>]>
+type StateModifier = {
+  $: 'state-modifier'
+  value: 'payload' | 'toggle' | ['increment', number]
+}
+
+export function isStateUpdateAction<T extends JSONValue>(
+  action: ActionDataState
+): action is UpdateStateAction<T> {
+  return action.name[0] === 'state-update'
+}
+
+export function state<T extends JSONValue>(initialValue: T): StateDataState<T> {
+  const key = (Date.now() * Math.random()).toString(16)
+  return { $: 'state', key, initialValue }
+}
+
+export function setStateAction<T>(
+  state: StateDataState<T>,
+  value: T | StateModifier = eventPayload
+): UpdateStateAction<T> {
+  return action(['state-update', state, value])
```

Comment:
I guess this will be the string convention of all our built in features, `kebab-case`? I think we used camelCase in other cases like goBack? Just want to make sure we have consistency

Reply:
Could be! I somewhat defaulted to kebab case and forgot Rise was camelCase. Technically, Rise is a separate project so it can have its own action names, but I think we should have a single way and I am fine with kebab case (refactor other actions in separate PR?)

Reply:
I am going to default to kebab-case and convert the rest in a follow-up